### PR TITLE
check panic during provision

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

**Comment by LiliDeng** [comment]
before
```
deployment failed. LisaException: OSProvisioningTimedOut: OS Provisioning for VM 'name' did not finish in the allotted time. The VM may still finish provisioning successfully. Please check provisioning state later. For details on how to check current provisioning state of Windows VMs, refer to https://aka.ms/WindowsVMLifecycle and Linux VMs, refer to https://aka.ms/LinuxVMLifecycle.
```

after
```
deployment failed. KernelPanicException: provision found panic in serial log: ['Synchronous Exception at 0x000000003FD04000\r']
```

**Comment by LiliDeng** [comment]
@squirrelsc please help check when you have time, it can give detailed reason if provision for panic, thanks!

File: lisa/util/__init__.py [hunk 1]
```diff
@@ -835,3 +855,22 @@ def get_first_combination(
         results.pop()
 
     return False
+
+
+def check_panic(content: str, stage: str, log: "Logger") -> None:
+    log.debug("checking panic in serial log...")
```
Comment (squirrelsc) [inline lisa/util/__init__.py]: The content may not from the serial log.

File: lisa/sut_orchestrator/azure/platform_.py [hunk 2]
```diff
@@ -678,7 +679,7 @@ def _delete_environment(self, environment: Environment, log: Logger) -> None:
             else:
                 log.debug("not wait deleting")
 
-    def _save_console_log(
+    def _check_panic(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: The method name should also specify it saves the serial log.

**Comment by squirrelsc** [comment]
> before
> 
> ```
> deployment failed. LisaException: OSProvisioningTimedOut: OS Provisioning for VM 'name' did not finish in the allotted time. The VM may still finish provisioning successfully. Please check provisioning state later. For details on how to check current provisioning state of Windows VMs, refer to https://aka.ms/WindowsVMLifecycle and Linux VMs, refer to https://aka.ms/LinuxVMLifecycle.
> ```
> 
> after
> 
> ```
> deployment failed. KernelPanicException: provision found panic in serial log: ['Synchronous Exception at 0x000000003FD04000\r']
> ```

"OSProvisioningTimedOut" should be in the error message too. So we know the provisioning is timed out, and we found kernel panic.

`deployment failed. LisaException: OSProvisioningTimedOut: KernelPanicException: provision found panic in serial log: ['Synchronous Exception at 0x000000003FD04000\r']`

**Comment by LiliDeng** [comment]
current exception message 
```
deployment failed. LisaException: OSProvisioningTimedOut: KernelPanicException: provision found panic in serial log: ['Synchronous Exception at 0x000000003FD04000\r']
```

File: lisa/util/__init__.py (hunk 1 above)
Comment (LiliDeng) [reply lisa/util/__init__.py]: updated

File: lisa/sut_orchestrator/azure/platform_.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/platform_.py]: updated

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**